sw/device/lib/crypto/include/hash.h

```diff
@@ -26,44 +26,28 @@ extern "C" {
  */
 typedef enum hash_mode {
   // SHA2-256 mode.
-  kHashModeSha256 = 0x783,
+  kHashModeSha256 = 0x69b,
   // SHA2-384 mode.
-  kHashModeSha384 = 0xd16,
+  kHashModeSha384 = 0x7ae,
   // SHA2-512 mode.
-  kHashModeSha512 = 0x2e9,
+  kHashModeSha512 = 0x171,
   // SHA3-224 mode.
-  kHashModeSha3_224 = 0x13d,
+  kHashModeSha3_224 = 0x516,
   // SHA3-256 mode.
-  kHashModeSha3_256 = 0xc99,
+  kHashModeSha3_256 = 0x2d4,
   // SHA3-384 mode.
-  kHashModeSha3_384 = 0xb58,
+  kHashModeSha3_384 = 0x267,
   // SHA3-512 mode.
```

Comment:
Out of curiosity, why are these values changing? Is there a tool that auto-populates these values?

Reply:
Yes, it's `/util/design/sparse-fsm-encode.py`. For a given number of values and bit-length of each value, it generates numbers with a high pairwise Hamming distance from each other -- very useful for this kind of thing! Since I combined some enums together that were previously generated separately, I needed to re-run it to generate a larger set to make sure the newly combined values had enough distance from each other.

Reply:
Thanks @jadephilipoom, I think the reason behind re-generation makes sense. For the RTL side, we always keep a comment header for this, which looks like below (I think this is also generated by the script):
```
  // Encoding generated with:
  // $ ./util/design/sparse-fsm-encode.py -d 5 -m 8 -n 10 \
  //      -s 4101887575 --language=sv
  //
  // Hamming distance histogram:
  //
  //  0: --
  //  1: --
  //  2: --
  //  3: --
  //  4: --
  //  5: |||||||||||||||||||| (50.00%)
  //  6: |||||||||||||| (35.71%)
  //  7: || (7.14%)
  //  8: || (7.14%)
  //  9: --
  // 10: --
  //
  // Minimum Hamming distance: 5
  // Maximum Hamming distance: 8
  // Minimum Hamming weight: 2
  // Maximum Hamming weight: 8
  //
```
Do you think it makes sense to also have them here as comment as well? If the complete Hamming weight depiction is cluttering the code too much, it might also suffice to just mention the parameters:
`$ ./util/design/sparse-fsm-encode.py -d 5 -m 8 -n 10 -s 4101887575 --language=sv`
I find the following information the most useful in this example:
```
  // Minimum Hamming distance: 5
  // Maximum Hamming distance: 8
  // Minimum Hamming weight: 2
  // Maximum Hamming weight: 8
```

Reply:
Yes, that style of comment appears everywhere internally, but not in the top-level cryptolib API; I didn't make that decision, but I'd guess the reasoning is that it would be too verbose or confusing to cryptolib users who aren't familiar with the OT-internal script. I think the idea of including just the invocation and the distance/weight parameters is a good one, without the histogram; this would also usually be enough information to generate additional values without changing existing ones. However, for consistency, I'd want to do this in the same style for _all_ hardened enums in the cryptolib API, so I've made an issue to re-generate all the enum values and add the comments: https://github.com/lowRISC/opentitan/issues/20051

Reply:
Thank you, sounds good.